experimental/segmenter/src/grapheme.rs

```diff
@@ -172,6 +172,8 @@ impl GraphemeClusterSegmenter {
     }
 
     /// Create a grapheme cluster break iterator from grapheme cluster rule payload.
+    ///
+    /// If the string is empty, no breakpoints are returned.
```

Comment:
Why?
If we always return 0 and length, then for an empty string we should get a list of one element `[0]`.
Let's please be consistent and predictable.

Reply:
Question: if we returns `[0]` for an empty string, does this violate the spec rules GB1 & GB2?

> GB1 & GB2: Break at the start and end of text, unless the text is empty.
https://www.unicode.org/reports/tr29/#Grapheme_Cluster_Boundary_Rules

Reply:
My reading of GB1 and GB2 is that we _have_ a break on the empty string, and the description is just wrong (the formulæ are normative, the descriptions are informative).

`sot` `eot` matches GB1 `sot` `÷`, so we get `sot` `÷` `eot`.

Reply:
ok I made it return `[0]` for the empty string. I made that change in both the UAX 14 and UAX 29 segmenters, right?

Reply:
Yes, `[0]` for the empty string from any segmenter. This is about predictable, usable API behavior.

Don't get hung up on what a standard says about these edge cases, especially when different specs give different guidance. If you want, you can document that the API always returns 0 and length even when some spec says one or the other is not a boundary in some sense.